packages/plugin-awesome/src/generators.ts

```diff
@@ -199,8 +200,12 @@ export const generateTree = async (
   tests: AwesomeTestResult[],
 ) => {
   const visibleTests = tests.filter((test) => !test.hidden);
+  const useCombined = labels.includes("titlePath");
+
   const tree: TreeData<AwesomeTreeLeaf, AwesomeTreeGroup> = labels.length
-    ? buildTreeByLabels(visibleTests, labels)
+    ? useCombined
```

Comment:
Let's avoid such deep ternary operators. Better to have a single switch/case expression or multiple if.